text/0095-prove-ownership.md

```diff
@@ -0,0 +1,239 @@
+
+**Should do:**
+
+Send message with TL-B schema to `dest` contract:
+```
+owner_info#c2fa9387 query_id:uint64 item_id:uint256 initiator:MsgAddress owner:MsgAddress 
+data:^Cell content:(Maybe ^Cell) = InternalMsgBody;
+```
+
+`query_id` - request number passed in `prove_ownership`.
+
+`item_id` -  id of NFT.
+
+`initiator` - address of request initiator.
+
+`owner` - current owner's address.
+
+`data` - data cell passed in `prove_ownership`.
+
+`content` - NFT's content, it is passed if `with_content` was true in `request_owner`.
+
+In case when `owner_info` was bounced back to NFT, NFT should send message to **initiator** with schema:
+```
+owner_info_bounced#7ca7b0fe query_id:uint64 item_id:uint256 initiator:MsgAddress owner:MsgAddress 
+data:^Cell content:(Maybe ^Cell) = InternalMsgBody;
+```
+
+### Implementation example
+https://github.com/getgems-io/nft-contracts/blob/main/packages/contracts/sources/sbt-item.fc
+
+# Guide
+
+#### Proving ownership to contracts
+You can send message to NFT and it will proxify message to target contract with its index and owner's wallet address in payload. 
+This way, target contract could know that you are owner of NFT that relates to expected collection. 
+Contract could know that NFT relates to collection by calculating address of NFT using code and index, and comparing it with sender.
+
+There are 2 methods which allow to use this functionality, **ownership proof** and **request owner info**. 
+The difference is that proof can be called only by NFT owner, so it is preferred to use when you need to accept messages only from owner, for example votes in DAO.
+
+##### Ownership proof
+**NFT owner** can send message to NFT with this schema:
+```
+prove_ownership#04ded148 query_id:uint64 dest:MsgAddress 
+forward_payload:^Cell with_content:Bool = InternalMsgBody;
+```
+After that NFT will send transfer to `dest` with scheme:
+```
+ownership_proof#6ecd55cc query_id:uint64 item_id:uint256 owner:MsgAddress 
+data:^Cell content:(Maybe ^Cell) = InternalMsgBody;
+```
+If something goes wrong and target contract not accepts message, and it will be bounced back to NFT, NFT will proxy this bounce to **owner**, this way coins will not stuck on NFT.
+Schema of message that will be send back to owner:
+```
+ownership_proof_bounced#c18e86d2 query_id:uint64 item_id:uint256 owner:MsgAddress 
+data:^Cell content:(Maybe ^Cell) = InternalMsgBody;
+```
+
+##### Owner info
+**anyone** can send message to NFT with this schema:
+```
+request_owner#d0c3bfea query_id:uint64 dest:MsgAddress 
+forward_payload:^Cell with_content:Bool = InternalMsgBody;
+```
+After that NFT will send transfer to `dest` with scheme:
+```
+owner_info#c2fa9387 query_id:uint64 item_id:uint256 initiator:MsgAddress owner:MsgAddress 
+data:^Cell content:(Maybe ^Cell) = InternalMsgBody;
+```
+If something goes wrong and target contract not accepts message, and it will be bounced back to NFT, NFT will proxy this bounce to **initiator**.
```

Comment:
```suggestion
If the target contract fails to process the message, it will be bounced back to NFT, NFT then should proxy that bounced message to the **initiator**.
```